converter.go

```diff
@@ -206,99 +210,129 @@ func snowflakeTypeToGoForMaps[K comparable](ctx context.Context, valueMetadata f
 
 // valueToString converts arbitrary golang type to a string. This is mainly used in binding data with placeholders
 // in queries.
-func valueToString(v driver.Value, tsmode snowflakeType) (*string, error) {
+func valueToString(v driver.Value, tsmode snowflakeType, params map[string]*string) (*string, string, *bindingSchema, error) {
```

Comment:
that's quite a lot of values that are being returned from this, looks like a red flag for me. Can't we encapsulate these somehow?

Reply:
Good idea, done.